libnmstate/nm/bond.py

```diff
@@ -53,12 +88,36 @@ def get_bond_info(nm_device):
 
 
 def get_options(nm_device):
-    con = connection.ConnectionProfile()
-    con.import_by_device(nm_device)
-    if con.profile:
-        bond_settings = con.profile.get_setting_bond()
-        return bond_settings.props.options
-    return {}
+    ifname = nm_device.get_iface()
+    if not ifname:
+        return {}
+
+    sysfs_folder = SYSFS_BOND_FOLDER.format(ifname)
+    options = {}
+    for sysfs_file_name in os.listdir(sysfs_folder):
+        if sysfs_file_name in NM_SUPPORTED_BOND_OPTIONS:
```

Comment:
Why this check is needed?

Reply:
There are many bond options in that folder which is not supported by NM.

Reply:
We can report them all nevertheless, why limit ourself?
When setting, an error will be given I guess.

Reply:
This will also remove the need to maintain the list.